rir/src/compiler/translations/rir_2_pir/rir_2_pir.cpp

```diff
@@ -1043,6 +1058,19 @@ Value* Rir2Pir::tryTranslate(rir::Code* srcCode, Builder& insert) const {
                 log.failed("Abort r2p due to unsupported bc");
                 return nullptr;
             }
+
+            if (!inPromise() && !insert.getCurrentBB()->isEmpty()) {
+                auto last = insert.getCurrentBB()->last();
+
+                if (Deopt::Cast(last)) {
```

Comment:
Isn't the break above on line 335 enough?

Reply:
no, break just leaves the compileBC  function. we also must abort all later instructions to avoid compiling dead code.